test/generators/AbstractGenerator.spec.ts

```diff
@@ -46,4 +55,22 @@ describe('AbstractGenerator', function() {
 
     expect(renderedContent).toEqual("SomeModel");
   });
+
+  describe('getPresets()', function() {
+    test('getPresets()', async function() {
+      class GeneratorWithPresets extends AbstractGenerator {
+        constructor() {
+          super("TestGenerator", {presets: [{"preset": {"test": "test2"}, "options": {}}]});
+        }
+        render(model: CommonModel, inputModel: CommonInputModel): any {
+          return model.$id || "rendered content";
+        }
+        testGetPresets(string: string){
+          return this.getPresets(string);
+        }
+      }
+      let newGenerator = new GeneratorWithPresets();
+      expect(newGenerator.testGetPresets("test")).toEqual([["test2", {}]]);
+    });
+  });
```

Comment:
@magicmatatjahu what this use-case is for? 🤔

Reply:
It gives you filtered presets by given name in array. It uses by generator to optimise retrieving presets :)